datafusion/sqllogictest/test_files/scalar.slt

```diff
@@ -1907,7 +1907,7 @@ select position('' in '')
 1
 
 
-query error DataFusion error: Execution error: The STRPOS/INSTR/POSITION function can only accept strings, but got Int64.
```

Comment:
i agree in principle, but actually why didn't this work?

Looking at the code here, every function that accepts a string, should be invocable with a number or pretty much anything
https://github.com/apache/datafusion/blob/8db30e25d6fe65a9779d237cf48aea9aee297502/datafusion/expr/src/type_coercion/functions.rs#L681-L682

i am not convinced this is how it should be, but that's not the point here. My question is -- why this didn't work.

cc @alamb @jayzhan211

Reply:
`coerce_types` doesn't go to `coerce_from`. That is the reason

Reply:
thanks, but not sure i understand.  It this explaining why the query fails before or after this PR?

Reply:
The query was already failing before this PR. The reason is in the function that computes the return type: 
```rust 
fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
        utf8_to_int_type(&arg_types[0], "strpos/instr/position")
    }
``` 
The source of the error is in these lines:
https://github.com/apache/datafusion/blob/5740774acc9febb51fafb17235066c4c5ef2b32a/datafusion/functions/src/utils.rs#L57-L67